brain/synapses.h

```diff
@@ -205,16 +205,25 @@ class Synapses
     BRAIN_API const float* decays() const;
 
     /** @return the absolute synaptic efficacies in millivolts. */
-    BRAIN_API const int* efficacys() const;
+    BRAIN_API const int* efficacies() const;
     //@}
 
+protected:
+    // The Impl pointer needs to be used in the Python wrapping as a custodian.
+    // Instead of moving the Impl declaration outside the implementation, this
+    // base class is used to give access to a virtual destructor, which is all
+    // the wrapping needs.
+    struct BaseImpl
+    {
+        virtual ~BaseImpl() {};
+    };
+    struct Impl;
```

Comment:
No, the forward declare is needed, otherwise you can't inherit Impl from BaseImpl unless BaseImpl is public, which looks ugly to me.
No the overhead doesn't matter too much, but do you really want me to spend time changing the code again?